src/libraries/Microsoft.Extensions.Diagnostics/src/DefaultMeterFactory.cs

```diff
@@ -0,0 +1,126 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+
+namespace Microsoft.Extensions.Metrics
+{
+    internal sealed class DefaultMeterFactory : IMeterFactory
+    {
+        private readonly Dictionary<string, List<Meter>> _cachedMeters = new();
+        private bool _disposed;
+
+        public DefaultMeterFactory() { }
+
+        public Meter Create(MeterOptions options)
+        {
+            if (options is null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            Debug.Assert(options.Name != null);
+
+            lock (_cachedMeters)
+            {
+                if (_disposed)
+                {
+                    throw new ObjectDisposedException(GetType().Name);
+                }
+
+                if (_cachedMeters.TryGetValue(options.Name, out List<Meter>? meterList))
```

Comment:
Is it correct that meter names are case sensitive?

Reply:
Yes, names are case sensitive.